automate-setup/automate-setup.md

```diff
@@ -0,0 +1,87 @@
+# Automate Setup
+
+## Problem
+
+At the moment, a completely hands-off setup of Quay post-install is difficult if not impossible. A typical automated setup would include creating users, teams, namespaces, etc. To do any of this, the first user must be created and the bearer token obtained for use with API calls.
+
+## Solution
+
+Create an unauthenticated API endpoint that will create the first user in the database, returning credentials for both pushing images as well as the bearer token for API use.
+
+## Implementation Details
+
+### Feature Flag
+
+Add a new feature `FEATURE_USER_INITIALIZE` which defaults to false.
+
+### API
+
+`/api/v1/user/initialize'
+
+Will create a new user iff there are no users in the database.
+
+Input `name`, `password`, `email`, and `access_token`. The first three are as described in regular user create API. The `access_token` is unique in that it is the scope of permissions to create an access token for this user.
+
+The scope may be one or more of the following in a single string separated by spaces:
```

Comment:
Is there a need for `scope`? As I understand the idea is to initialize the `admin` user which can later be used to initialize other users. If someone creates a user with `user:read` scope only then you'll have to manually create the admin user.

Reply:
I originally just had `access_token: true` which would then have all permissions added to the token. I'm can't think of a reason not to give all permissions.